src/lib/OS/SimOS.cc

```diff
@@ -8,12 +10,19 @@ namespace OS {
 
 SimOS::SimOS(std::string executablePath,
              std::vector<std::string> executableArgs,
-             std::shared_ptr<simeng::memory::Mem> mem)
+             std::shared_ptr<simeng::memory::Mem> mem, bool setProcess)
     : executablePath_(executablePath),
       executableArgs_(executableArgs),
-      memory_(mem),
-      syscallHandler_(std::make_shared<SyscallHandler>(processes_)) {
-  createInitialProcess();
+      memory_(mem) {
+  syscallHandler_ = std::make_shared<SyscallHandler>(this);
+  // Callback function used to send data to memory on page fault. This was
+  // preferred to a sharing the memory class as this callback can now be
+  // implementation specific.
+  sendToMem_ = [&, this](std::vector<char> data, uint64_t addr, size_t size) {
+    this->memory_->sendUntimedData(data, addr, size);
+  };
```

Comment:
Capture list need only contain `[this]`, and can remove `this->` from the function body.

Also, could `sendToMem_` not be sent to the process on construction where it is stored as a member variable? Then, we can eliminate the use of `memory` in the Process' constructor, and don't need to pass the function into the `handlePageFault` function each time

Reply:
Any thoughts on the second part of this?

Reply:
Yes that makes sense, sorry it skipped my mind. Will make that change now